File: Professional Image Modifier 5/[VB]-Professional Image Modifier 5/VB/ImageFunctions/Forms/FrmCommandBox.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Imports WeifenLuo.WinFormsUI.Docking


Namespace ImageFunctions.Forms
	Partial Public Class FrmCommandBox
		Inherits DockContent

		Public Sub New()
			InitializeComponent()
		End Sub


		Protected Overrides Function GetPersistString() As String
			Return Me.Text
		End Function

		''' <summary>
		''' This produced the flash required to attract attention to this being an active input box
		''' </summary>
		''' <param name="sender"></param>
		''' <param name="e"></param>
		Private Sub timerFlash_Tick_1(ByVal sender As Object, ByVal e As EventArgs) Handles timerFlash.Tick
			If lblFlash.Text = ":" Then
				lblFlash.Text = " "
			Else
				lblFlash.Text = ":"
			End If


		End Sub
	End Class
End Namespace

File: Professional Image Modifier 5/[C#]-Professional Image Modifier 5/C#/ImageFunctions/Forms/FrmCommandBox.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

using WeifenLuo.WinFormsUI.Docking;


namespace ImageFunctions.Forms
{
    public partial class FrmCommandBox : DockContent
    {
        public FrmCommandBox()
        {
            InitializeComponent();
        }


        protected override string GetPersistString()
        {
            return this.Text;
        }

        /// <summary>
        /// This produced the flash required to attract attention to this being an active input box
        /// </summary>
        /// <param name="sender"></param>
        /// <param name="e"></param>
        private void timerFlash_Tick_1(object sender, EventArgs e)
        {
            if (lblFlash.Text == ":")
                lblFlash.Text = " ";
            else
                lblFlash.Text = ":";

           
        }
    }
}
